File: Practica2-MascotasUN/src/gestorAplicacion/gestionVentas/Producto.py
# Clase que crea y gestiona los objetos de tipo Producto, los cuales son los productos que la aplicación vende
from  random import randint
class Producto:
    productos = {}
    productoID = randint(444444,888888)

    # Constructor
    def __init__(self, nombre, fechaVencimiento, descripcion, precioCompra, precioVenta, cantidadComprada, raza="",
                 etapa="", especie=""):
        Producto.productoID+=1
        self.productoID=Producto.productoID
        self.nombre = nombre
        self.fechaVencimiento = fechaVencimiento
        self.descripcion = descripcion
        self.precioCompra = precioCompra
        self.precioVenta = precioVenta
        self.cantidadComprada = cantidadComprada
        self.raza = raza
        self.etapa = etapa
        self.especie = especie
        Producto.productos[self.productoID] = self

    # Métodos

    # Método sobrecargado toString para imprimir los datos de un producto
    def __str__(self) -> str:
        return "Código: " + str(self.productoID) + "\n" \
               + "Producto: " + str(self.nombre) + "\n" \
               + "Fecha de Vencimiento: " + str(self.fechaVencimiento) + "\n" \
               + "Descripción: " + str(self.descripcion) + "\n" \
               + "Precio: " + str(self.precioVenta) + "\n"

    # Método encargado de editar los atributos de un producto
    def editarProducto(self, nombre, precioVenta, cantidadComprada):
        self.nombre = nombre
        self.precioVenta = precioVenta
        self.cantidadComprada = cantidadComprada

    @staticmethod
    def crearObjeto(nombre, fechaVencimiento, descripcion, precioCompra, precioVenta, cantidadComprada, raza="",
                    etapa="", especie=""):
        try:
            precioCompra = int(precioCompra)
            precioVenta = int(precioVenta)
            cantidadComprada = int(cantidadComprada)
            Producto(nombre, fechaVencimiento, descripcion, precioCompra, precioVenta, cantidadComprada, raza, etapa,
                     especie)
            return True
        except ValueError:
            return False

    # Método que elimina un producto de la lista de productos a comprar
    @classmethod
    def eliminarProducto(cls, productoID):
        for producto in cls.productos:
            if producto.productoID == productoID:
                cls.productos.remove(producto)
                return True
        return False

    # Método que agrega un producto a la lista de productos a comprar
    @classmethod
    def agregarProducto(cls, producto):
        cls.productos.append(producto)

    # Setters y getters
    @classmethod
    def getProductoId(cls):
        return cls.productoID

    @classmethod
    def setProductoId(cls, productoID):
        cls.productoID = productoID

    @classmethod
    def getProductos(cls):
        return cls.productos

    @classmethod
    def setProductos(cls, productos):
        cls.productos = productos

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def getFechaVencimiento(self):
        return self.fechaVencimiento

    def setFechaVencimiento(self, fechaVencimiento):
        self.fechaVencimiento = fechaVencimiento

    def getDescripcion(self):
        return self.descripcion

    def setDescripcion(self, descripcion):
        self.descripcion = descripcion

    def getPrecioCompra(self):
        return self.precioCompra

    def setPrecioCompra(self, precioCompra):
        self.precioCompra = precioCompra

    def getPrecioVenta(self):
        return self.precioVenta

    def setPrecioVenta(self, precioVenta):
        self.precioVenta = precioVenta

    def getCantidadComprada(self):
        return self.cantidadComprada

    def setCantidadComprada(self, cantidadComprada):
        self.cantidadComprada = cantidadComprada

    def getRaza(self):
        return self.raza

    def setRaza(self, raza):
        self.raza = raza

    def getEtapa(self):
        return self.etapa

    def setEtapa(self, etapa):
        self.etapa = etapa

    def getEspecie(self):
        return self.especie

    def setEspecie(self, especie):
        self.especie = especie

File: Practica1-MascotasUN/src/main/java/gestorAplicacion/gestionVentas/Producto.java
/*Clase que crea y gestiona los objetos de tipo Producto, los cuales son los productos que la aplicación vende*/

package gestorAplicacion.gestionVentas;

import java.io.Serializable;
import java.util.ArrayList;

public class Producto implements Serializable {

	//Serializador Productos
	private static final long serialVersionUID = 1L;
    private static ArrayList<Producto> productos = new ArrayList<>();
    
    //Atributos
    private long productoID;
    private String nombre;
    private String fechaVencimiento;
    private String descripcion;
    private float precioCompra;
    private float precioVenta;
    private int cantidadComprada;
    private String raza;
    private String etapa;
    private String especie;
    //Lista para la recomendacion de concentrado, anadir tipo. y Sobrecarga de constructores.

    // Constructor
    public Producto(long productoID, String nombre, String fechaVencimiento, String descripcion, float precioCompra, float precioVenta, int cantidadComprada) {
	    this.productoID = productoID;
	    this.nombre = nombre;
	    this.fechaVencimiento = fechaVencimiento;
	    this.descripcion = descripcion;
	    this.precioCompra = precioCompra;
	    this.precioVenta = precioVenta;
	    this.cantidadComprada = cantidadComprada;
        productos.add(this);
    }
    
  //Sobreescritura de constructor para crear productos relacionados con gatos
    public Producto(long productoID, String nombre, String fechaVencimiento, String descripcion, float precioCompra,
			float precioVenta, int cantidadComprada, String etapa, String especie) {
		this.productoID = productoID;
		this.nombre = nombre;
		this.fechaVencimiento = fechaVencimiento;
		this.descripcion = descripcion;
		this.precioCompra = precioCompra;
		this.precioVenta = precioVenta;
		this.cantidadComprada = cantidadComprada;
		this.etapa = etapa;
		this.especie = especie;
        productos.add(this);
	}
    
    //Sobreescritura de constructor para crear productos relacionados con perros
    public Producto(long productoID, String nombre, String fechaVencimiento, String descripcion, float precioCompra,
			float precioVenta, int cantidadComprada, String raza, String etapa, String especie) {
		this.productoID = productoID;
		this.nombre = nombre;
		this.fechaVencimiento = fechaVencimiento;
		this.descripcion = descripcion;
		this.precioCompra = precioCompra;
		this.precioVenta = precioVenta;
		this.cantidadComprada = cantidadComprada;
		this.raza = raza;
		this.etapa = etapa;
		this.especie = especie;
        productos.add(this);
	}

    // Métodos
    
    // Metodo sobrecargado toString() para generar el string que se imprimira al implementar la UI
    @Override
    public String toString() {
        return "~ Codigo = " + productoID +"\n"+
                "~ Producto = " + nombre + '\n' +
                "~ fechaDeVencimiento =" + fechaVencimiento +"\n"+
                "~ Descripcion = " + descripcion+ "\n"+
                "~ Precio =" + precioVenta;
    }
    
    // Método encargado de editar los atributos de un producto
    public void editarProducto(long productoID,String nombre,int precio, int cantidadComprada){
        this.nombre=nombre;
        this.productoID=productoID;
        this.precioVenta = precio;
        this.cantidadComprada = cantidadComprada;
    }
    
    // Método que elimina un producto de la lista de productos a comprar
    public static boolean eliminarProducto(Producto producto){
        Tienda.getCatalogo().remove(producto);
        Producto.getProductos().remove(producto);
        return true;
    }
    
    // Método que agrega un producto a la lista de productos a comprar
    public static void agregarProducto(Producto producto){
        Producto.productos.add(producto);
    }

    //getter y setter del serializador y deserializador
    public static ArrayList<Producto> getProductos() {
        return productos;
    }

    public static void setProductos(ArrayList<Producto> productos) {
        Producto.productos=productos;
    }

    // Getters y setters
    
    public long getProductoID() {
        return productoID;
    }

    public void setProductoID(long productoID) {
        this.productoID = productoID;
    }
    
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public String getFechaVencimiento() {
        return fechaVencimiento;
    }

    public void setFechaVencimiento(String fechaVencimiento) {
        this.fechaVencimiento = fechaVencimiento;
    }
    
    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
    
    public float getPrecioCompra() {
		return precioCompra;
	}

	public void setPrecioCompra(float precioCompra) {
		this.precioCompra = precioCompra;
	}

	public float getPrecioVenta() {
		return precioVenta;
	}

	public void setPrecioVenta(float precioVenta) {
		this.precioVenta = precioVenta;
	}

	public int getCantidadComprada() {
        return cantidadComprada;
    }

    public void setCantidadComprada(int cantidadComprada) {
        this.cantidadComprada = cantidadComprada;
    }

	public String getRaza() {
		return raza;
	}

	public void setRaza(String raza) {
		this.raza = raza;
	}

	public String getEtapa() {
		return etapa;
	}

	public void setEtapa(String etapa) {
		this.etapa = etapa;
	}

	public String getEspecie() {
		return especie;
	}

	public void setEspecie(String especie) {
		this.especie = especie;
	}
}